pkg/clusterautoscaler/flags.go

```diff
@@ -945,3 +953,50 @@ func PrefillAutoscalerArgs(cmd *cobra.Command, autoscalerArgs *AutoscalerArgs,
 	}
 	return autoscalerArgs, nil
 }
+
+// lassic clusters only
+func getAutoscalerMaxNodesTotalDefaultValue(
+	prefilledMaxNodesTotalValue int, autoscalerValidationArgs *AutoscalerValidationArgs) int {
+	if prefilledMaxNodesTotalValue > 0 {
+		return prefilledMaxNodesTotalValue
+	}
+
+	workerNodeCount := 180
+	infraNodeCount := 2
+	masterNodeCount := 3
+
+	classicMaxNodeSize249SupportedVersion, err := versions.IsGreaterThanOrEqual(
+		autoscalerValidationArgs.ClusterVersion, ocm.ClassicMaxNodeSize249Support)
+	if err != nil {
+		return workerNodeCount + infraNodeCount + masterNodeCount
+	}
+
+	if classicMaxNodeSize249SupportedVersion {
+		workerNodeCount = 249
+	}
+	if autoscalerValidationArgs.MultiAz {
+		infraNodeCount = 3
+	}
+
+	return workerNodeCount + infraNodeCount + masterNodeCount
+}
+
+// classic clusters only
+func validateAutoscalerMaxNodesTotal(autoscalerValidationArgs *AutoscalerValidationArgs) interactive.Validator {
+	return func(val interface{}) error {
+		maxNodesTotalDefaultValue := getAutoscalerMaxNodesTotalDefaultValue(0, autoscalerValidationArgs)
+		number, err := strconv.Atoi(fmt.Sprintf("%v", val))
+		if err != nil {
+			return fmt.Errorf("should provide an integer number between 0 to 2147483647")
```

Comment:
I think go has a built-in value for the max int value allowed (`math.MaxInt`) - I see this number is used a couple times in this MR is why I bring it up :) 

Hope I'm not being too nitpicky with some of my comments

Reply:
Unless there's a specific max we need here, like if we are converting to an int16 or int8 or something in CS (not sure if there's a constant value for a max int8 🤔 )